From a6ec6abcf585063b53311d1d008c3c3fafbb26ed Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 25 Jan 2024 17:36:00 +0100
Subject: [PATCH 0866/1141] drm/stm: dsi: remove loopback pixel clock

Parent of pixel clock should be set by the display controller.

Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Change-Id: I987af761c32449baaa8d7a36ba7fedf8068af014
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/354667
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/stm/dw_mipi_dsi-stm.c | 15 ---------------
 1 file changed, 15 deletions(-)

--- a/drivers/gpu/drm/stm/dw_mipi_dsi-stm.c
+++ b/drivers/gpu/drm/stm/dw_mipi_dsi-stm.c
@@ -135,7 +135,6 @@ struct dw_mipi_dsi_stm {
 	struct clk *pllref_clk;
 	struct clk *pclk;
 	struct clk *px_clk;
-	struct clk *ltdc_clk;
 	struct clk_hw txbyte_clk;
 	struct dw_mipi_dsi *dsi;
 	u32 hw_version;
@@ -390,13 +389,6 @@ static int dw_mipi_dsi_phy_141_init(void
 
 	DRM_DEBUG_DRIVER("\n");
 
-	/* Switch pixel clock parent to ref clock */
-	ret = clk_set_parent(dsi->ltdc_clk, dsi->px_clk);
-	if (ret) {
-		DRM_ERROR("Could not set pixel clock parent: %d\n", ret);
-		return ret;
-	}
-
 	/* Select video mode by resetting DSIM bit */
 	dsi_clear(dsi, DSI_WCFGR, WCFGR_DSIM);
 
@@ -1261,13 +1253,6 @@ static int dw_mipi_dsi_stm_probe(struct
 	if (IS_ERR(dsi->px_clk))
 		dev_err_probe(dev, PTR_ERR(dsi->px_clk), "Unable to get px_clk clock\n");
 
-	if (dsi->hw_version == HWVER_141) {
-		dsi->ltdc_clk = devm_clk_get(dev, "pixclk");
-		/* No need to return since only MP25 has it */
-		if (IS_ERR(dsi->ltdc_clk))
-			dev_err_probe(dev, PTR_ERR(dsi->ltdc_clk), "Unable to get pixclk clock\n");
-	}
-
 	dsi->probe_done = true;
 
 	/*
